ci-operator/step-registry/openshift-qe/rosa-loaded-upgrade/openshift-qe-rosa-loaded-upgrade-commands.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+# Log in
+function rosa_login()
+{
+  if [ $# -eq 1 ];then
+    REGION=$1
+  fi
+
+  ROSA_LOGIN_ENV=${ROSA_LOGIN_ENV:="staging"}
+  ROSA_VERSION=$(rosa version)
+  ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+
+  if [[ ! -z "${ROSA_TOKEN}" ]]; then
+     echo "Logging into ${ROSA_LOGIN_ENV} with offline token using rosa cli ${ROSA_VERSION}"
+     rosa login --env "${ROSA_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+     if [ $? -ne 0 ]; then
+       echo "Login failed"
+       exit 1
+     fi
+  else
+     echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+     exit 1
+  fi
+  echo
+  echo "######################################################################"
+  rosa list clusters --region $REGION
+  echo "######################################################################"
+  echo 
+}
+
+function classic_rosa_upgrade()
+{
+
+  SCHEDULE_OFFSET=15  #After $SCHEDULE_OFFSET minutes to upgrade
+  SCHEDULE_DATETIME=`date  -d "+${SCHEDULE_OFFSET} min" "+%Y-%m-%d %H:%M"`
+  SCHEDULE_DATE=$(echo $SCHEDULE_DATETIME | awk '{print $1}')
+  SCHEDULE_TIME=$(echo $SCHEDULE_DATETIME | awk '{print $2}')
+
+  RELEASE_IMAGE_LATEST=${RELEASE_IMAGE_LATEST:=""}
+  CURRENT_VERSION=$(oc get clusterversion -ojsonpath={..desired.version})
+  OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE:=$RELEASE_IMAGE_LATEST}
+
+  SHARED_DIR=${SHARED_DIR:=""}
+  if [ $# -eq 1 ];then
+    REGION=$1
+  fi
+
+  CLUSTER_NAME=${CLUSTER_NAME:=""}
+  if [[ -z $CLUSTER_NAME && -s "${SHARED_DIR}/cluster-name" ]];then
+        CLUSTER_NAME=$(cat ${SHARED_DIR}/cluster-name)
```

Comment:
We will always have access to CLUSTER_ID. IMO, if you use CLUSTER_ID instead of CLUSTER_NAME, this whole section can be removed. You can pass CLUSTER_ID to `rosa upgrade`

Reply:
I tried to use CLUSTER_ID directly, but fail to get the value of CLUSTER_ID in rosa-pgrade steps, so, still need to check if the file exist, but I use CLUSTER_ID instead of CLUSTER_NAME